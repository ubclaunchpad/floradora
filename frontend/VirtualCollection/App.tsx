import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import 'react-native-gesture-handler';
import CameraPage from './src/pages/CameraPage';
import CollectionPage from './src/pages/CollectionPage';
import HomePage from './src/pages/HomePage';
import LoginPage from './src/pages/LoginPage';
import { AppStackParamList } from './src/pages/types';

export default function App() {
    const Stack = createStackNavigator<AppStackParamList>();
    return (
        <NavigationContainer>
            <Stack.Navigator>
                <Stack.Screen name="Login" component={LoginPage} />
                <Stack.Screen name="Home" component={HomePage} />
                <Stack.Screen name="Camera" component={CameraPage} />
                <Stack.Screen name="Collection" component={CollectionPage} />
            </Stack.Navigator>
        </NavigationContainer>
    );
}
