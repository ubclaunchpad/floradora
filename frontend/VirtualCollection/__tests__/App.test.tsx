import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('expo-linking');

describe('<App />', () => {
    it('matches snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
