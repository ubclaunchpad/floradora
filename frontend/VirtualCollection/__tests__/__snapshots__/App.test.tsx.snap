// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.js to start working on your app!
  </Text>
</View>
`;
